### PR TITLE
Remove exceptions suppress

### PR DESCRIPTION
**Pull request by torbacz**
## Description

-  Removed exceptions suppressions from code (and null returning when data is invalid)
- Fixed tests

## Motivation and Context

- Fixes nanoFramework/Home#1114

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [x] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [x] I have added new tests to cover my changes.

### CODE REVIEW

**Review by josesimoes — COMMENTED** [review]
Despite lots of changes LGTM. 😅 
Unit tests are all green which is good too.

Just a nit pick on the comment.

File: nanoFramework.Json.Test/JsonUnitTests.cs [hunk 1]
```diff
@@ -436,19 +436,17 @@ public void Can_serialize_deserialize_timespan_02()
...
-                var dserResult = (JsonTestClassTimeSpan)JsonConvert.DeserializeObject(strArr[i], typeof(JsonTestClassTimeSpan));
-
-                endTimestamp = Environment.TickCount64;
-                OutputHelper.WriteLine($"Deserialization took {endTimestamp - startTimestamp}ms");
+                try
+                {
+                    var _ = (JsonTestClassTimeSpan)JsonConvert.DeserializeObject(strArr[i], typeof(JsonTestClassTimeSpan));
+                    throw new Exception($"Should throw exception {nameof(InvalidCastException)}.");
+                }
+                catch (InvalidCastException) // Should throw InvalidCaseException
```
[the remark this answers is not in the document]
Comment (torbacz) [reply nanoFramework.Json.Test/JsonUnitTests.cs]: You are absolutely right!

**Comment by torbacz** [comment]
@josesimoes 
Not sure what happed there, probable IDE just reformatted whole code in "JsonConverter" class after removing try/catch block. Haven't noticed that. In fact, there are only three changes:

- Removing try/catch block from JsonConverter.PopulateObject method
- Replaced "return null" in JsonConverter.PopulateObject method with "throw new DeserializeException"
- Adjusted TimeSpanExtensions.TryConvertFromString to throw InvalidCastException when wrong data provided (instead of returning false)

**Review by josesimoes — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**